server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/FilterAllocationDecider.java

```diff
@@ -102,6 +102,17 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
         return shouldFilter(shardRouting, node.node(), allocation);
     }
 
+    @Override
+    public Decision canForceDuringVacate(ShardRouting shardRouting, RoutingNode node, RoutingAllocation allocation) {
+        if (replacementFromSourceToTarget(allocation.metadata(), shardRouting.currentNodeId(), node.nodeId())) {
+            // Allow overriding this decider during node replacement
+            return Decision.single(Decision.Type.YES, NAME, "overriding allocation filters during node replacement of [%s] with [%s]",
```

Comment:
A detail, perhaps we want to let the caller pass in part of the reason or append it afterwards, "node replacement of [a] with [b]"? Would be nice to have a consistent message.